accounts-db/src/tiered_storage.rs

```diff
@@ -115,19 +115,23 @@ impl TieredStorage {
             ));
         }
 
-        let result = {
-            let writer = TieredStorageWriter::new(&self.path, format)?;
-            writer.write_accounts(accounts, skip)
-        };
+        if format == &HOT_FORMAT {
+            let result = {
+                let writer = HotStorageWriter::new(&self.path)?;
+                writer.write_accounts(accounts, skip)
+            };
+
+            // panic here if self.reader.get() is not None as self.reader can only be
+            // None since we have passed `is_read_only()` check previously, indicating
+            // self.reader is not yet set.
+            self.reader
+                .set(TieredStorageReader::new_from_path(&self.path)?)
+                .unwrap();
```

Comment:
So in the upper layer, AccountsFile::append_accounts returns an Option instead.  This means when we have an error here, it's either panic or we hide the error.

I can probably improve the comment a bit by saying that it is guaranteed to have no reader here as we just checked its is_read_only() so we expect `reader.set()` to be successful.